tests/runtime/engine/runner.py

```diff
@@ -108,7 +108,7 @@ def prepare_bpf_call(test, nsenter=[]):
 
             return nsenter_prefix + ret
         else:  # PROG
-            use_json = "-q -f json" if test.expects[0].mode == "json" else ""
+            use_json = "-q -f json" if (not test.has_return_code and test.expects[0].mode == "json") else ""
```

Comment:
```suggestion
            use_json = "-q -f json" if (len(test.expects) > 0 and test.expects[0].mode == "json") else ""
```
To allow using both `RETURN_CODE` and `EXPECT_JSON` at the same time